script/LaunchProjectAndDeployDomJuice.s.sol

```diff
@@ -0,0 +1,145 @@
+//import {LaunchProjectData} from "./structs/LaunchProjectData.sol";
+
+contract LaunchAndDeploy is Script {
+    // Juicebox contracts
+    struct Contracts {
+        IJBController3_1 controller; // Controller that configures funding cycles
+        IJBSingleTokenPaymentTerminalStore3_1_1 paymentTerminalStore; // Stores all payment terminals
+        JBETHPaymentTerminal3_1_1 ethPaymentTerminal; // Default ETH payment terminal
+    }
+
+    Contracts contracts;
+
+    // JB Project Launch struct array parameters
+    IJBPaymentTerminal[] _terminals; // default empty
+    JBGroupedSplits[] _groupedSplits; // Default empty
+    JBFundAccessConstraints[] _fundAccessConstraints; // Default empty
+
+    // Campaign parameters
+    address payable owner = payable(msg.sender);
+    uint256 cycleTarget = 1200 ether;
+    uint256 startTime = block.timestamp + 172800; // 48 hours from script execution
+    uint256 duration = 1209600; // Two weeks of seconds
+    uint256 minimumPledgeAmount = 50 ether;
+
+    event DelegateDeployed(uint256 _projectID, address _delegate, address _owner);
+
+    function run() external returns (uint256 projectID) {
+        // Store the deployed JB contracts depending on which network (Mainnet or Goerli) is called for
+        HelperConfig helperConfig = new HelperConfig();
+        (address _controller, address _paymentTerminalStore3_1_1, address _ethPaymentTerminal3_1_1) =
+            helperConfig.activeNetworkConfig();
+        contracts.controller = IJBController3_1(_controller);
+        contracts.paymentTerminalStore = IJBSingleTokenPaymentTerminalStore3_1_1(_paymentTerminalStore3_1_1);
+        contracts.ethPaymentTerminal = JBETHPaymentTerminal3_1_1(_ethPaymentTerminal3_1_1);
+
+        _terminals.push(contracts.ethPaymentTerminal);
+
+        string memory ipfsCID = "QmZpzHK5tuNwVkm2EyJp2tVraD6xSJqdF2TE39hzinr9Bs"; // Project's metadata file hash and codec from IPFS (placeholder)
+
+        vm.startBroadcast();
+        // Launch the JB project and deploy the DAC delegate.
+        projectID = launchProjectFor(owner, ipfsCID, cycleTarget, startTime, duration, minimumPledgeAmount);
+        vm.stopBroadcast();
+    }
+
+    /// @notice Launches a new project with the DAC delegate (DominantJuice.sol) attached.
+    /// @param _owner The address to set as the owner of the project. The project's ERC-721 will be owned by this address.
+    /// @param _ipfsCID The CID obtained from IPFS for uploading project's metadata (name, logo, description, etc).
+    /// @param _cycleTarget Funding target for the dominant assurance campaign.
+    /// @param _startTime The timestamp that the campaign cycle is desired to start.
+    /// @param _duration The duration of the campaign cycle in seconds.
+    /// @param _minimumPledgeAmount Minimum amount that payers can pledge.
+    /// @return _projectID The ID of the newly configured project.
+    function launchProjectFor(
```

Comment:
I prefer structs vs this many individual parameters. It makes it a lot easier / safer to change later as well

Reply:
I guess how would you populate the struct with user-defined parameters then, if we would use the CLI? There's not struct syntax that I'm aware of.

Reply:
hmmm....maybe forge has an "args parser" concept? I mean, if you're supposed to write scripts in solidity, you need to be able to parse args from the command-line and have help scripts, etc. Otherwise, you need a wrapper script in another language that does all of those nice-to-have script things and just stop using solidity to write scripts in the first place.

Reply:
btw, that's something I would expect the `run` function to do....read the args from the command-line and pass them into `launchProjectFor` with all the parameters in a nice structured way.